content/spire/try/oidc-federation-aws.md

```diff
@@ -0,0 +1,415 @@
+
+In the YAML files, instances of the `example.org` [trust domain](/spiffe/concepts/#trust-domain) are valid to use for this tutorial and do not need to be changed.
+
+## Deploy OIDC Discovery Provider Configmap
+
+The SPIRE OIDC Discovery Provider provides a URL to the location of the discovery document specified by the OIDC protocol. The oidc-dp-configmap.yaml file specifies the URL to the OIDC Discovery Provider.
+
+Before running the command below, ensure that you have replaced the MY\_DISCOVERY\_DOMAIN placeholder with the FQDN of the Discovery Provider as described in [Replace Placeholder Strings in YAML Files](#replace-placeholder-strings-in-yaml-files).
+
+Change to the directory containing the **k8s/oidc-aws** YAML files and use the following command to apply the updated server configmap, the configmap for the OIDC Discovery Provider, and deploy the updated **spire-server** statefulset:
+
+```console
+$ kubectl apply \
+    -f server-configmap.yaml \
+    -f oidc-dp-configmap.yaml \
+    -f server-statefulset.yaml
+```
+
+To verify that the spire-server-0 pod has spire-server and spire-oidc containers, run:
+
+```console
+$ kubectl get pods spire-server-0 -n spire -o jsonpath='{.spec.containers[*].name}{"\n"}'
+```
+
+## Configure the OIDC Discovery Provider Service and Ingress
+
+Use the following command to set up a service definition for the OIDC Discovery Provider and to configure ingress for that service:
+
+```console
+$ kubectl apply \
+    -f server-oidc-service.yaml \
+    -f ingress.yaml 
+```
+
+## Verify the spire-server Stateful Set
+
+Verify that the spire-server stateful set has been created:
+
+```console
+$ kubectl get statefulset --namespace spire
+
+NAME           READY   AGE
+spire-server   0/1     10h
+```
+
+## Verify That spire-oidc has an External Service Address
+
+Run the following command to verify that the EXTERNAL-IP value is present for the spire-oidc service. The spire-oidc Discovery Provider service must provide an external IP address for AWS to access the OIDC discovery document provided by spire-oidc.
+
+```console
+$ kubectl get service -n spire
+
+NAME           TYPE           CLUSTER-IP    EXTERNAL-IP    PORT(S)          AGE
+spire-oidc     LoadBalancer   10.12.0.18    34.82.139.13   443:30198/TCP    108s
+spire-server   NodePort       10.12.13.51   <none>         8081:32115/TCP   10h
+```
+
+### Configure a DNS A Record to Point to the spire-oidc EXTERNAL-IP
+
+At this point you can put the value of the spire-oidc EXTERNAL-IP in the required DNS A record, along with the value of the MY\_DISCOVERY\_DOMAIN domain that you specified in oidc-dp-configmap.yaml. For more details, see [Required DNS A Record for the OIDC Discovery Provider IP Address](#required-dns-a-record-for-the-oidc-discovery-provider-ip-address).
```

Comment:
Particularly since there are propagation delays, we should probably ask the user to verify their DNS entry at this point, using something like `nslookup` before proceeding

Reply:
Done, covered in new part 2.